js/wallet.js

```diff
@@ -5,25 +5,89 @@
 'use strict'
 
 const Immutable = require('immutable')
-
 const sinon = require('sinon')
+
 const Wallet = require('../abs/wallet')
 const {request, roundtrip} = require('../lib/request')
+const settings = require('../browser-laptop/js/constants/settings')
+const responses = require('../lib/data/responses.json')
 
 const defaultAppState = Immutable.fromJS({
   cache: {
     ledgerVideos: {}
   },
-  ledger: {}
+  ledger: {
+    about: {}
+  },
+  about: {
+    preferences: {}
+  }
 })
 
 class JS extends Wallet {
   constructor () {
     super()
     this.ledger = null
+    this.minimumVisits = 1
+    this.minimumVisitTime = 5000
+    this.defaultContribution = 10
+    this.paymentsEnabled = false
+    this.state = defaultAppState
+    this.stateFile = null
+    this.wallet = 'default-wallet'
+    this.stateKey = 'default-state'
+  }
+
+  setStateFile () {
+    const ledgerStateData = require('../lib/data/ledger-state.json')
+    this.stateFile = JSON.stringify(ledgerStateData[this.stateKey])
+  }
+
+  setState (newState) {
+    const oldStateJS = this.state.toJS()
+    const newStateJS = newState.toJS()
+    this.state = newState
```

Comment:
Why do we have it here as plain JS and on line 73 as Immutable? Can we have it as Immutable all the time?

Reply:
Good catch, this should be immutable

Reply:
great, then we can remove all this toJS and fromJS conversions